MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -348,15 +359,21 @@ void EnggDiffFittingPresenter::processMultiRun(
 void EnggDiffFittingPresenter::processSingleRun(
     const std::string &focusDir, const std::string &strFocusedFile,
     std::vector<std::string> &runnoDirVector,
-    const std::vector<std::string> &splitBaseName,
-    std::vector<std::string> &runNoVec) {
+    const std::vector<std::string> &splitBaseName) {
+
+  if (g_fitting_runno_counter == 0) {
+    g_multi_run_directories.clear();
+  }
 
   // to track the FittingRunnoChanged loop number
   g_fitting_runno_counter++;
 
   m_view->setFittingSingleRunMode(true);
 
-  updateFittingDirVec(focusDir, strFocusedFile, runnoDirVector);
+  // dummy vector will not be used
+
+  std::vector<std::string> foundRunNumber;
+  updateFittingDirVec(focusDir, strFocusedFile, runnoDirVector, foundRunNumber);
```

Comment:
Could you create it inside the function only if it finds something, so in the case of nothing found you could return a nullptr and then check for that later?